### DISCUSSION TITLE
Skin tones and Github Emojis (:octocat:, :trollface:)

### DISCUSSION THREAD

**Discussion by 88572** [vercte/forumoji#58] [Ideas]
I was wondering if we could get skin tones? I couldn't make them, sorry. (and :octocat:)

**Comment by cocoelacanth** [comment]
Skin tone support is possible, but there are quite a few considerations to be made.

Forumoji currently doesn't support skin tones, as these are stored [in their own list](https://unicode.org/emoji/charts/full-emoji-modifiers.html) separate from the main one. Thankfully, it shouldn't be too difficult to add these.

Next, there's the matter of scope. Currently, the emoji specification defines [357 emojis](https://unicode.org/emoji/charts/emoji-counts.html) that have skin tone variants (1785 total variants divided by 5 skin tones). Of these 357 emojis, Forumoji currently has 113, meaning that we need 565 new emojis to cover all the skin tones.

In addition to these 113 emojis, we'll need some way of dealing with the remaining 244 that haven't been drawn yet. I can see two possible strategies for this:
* **add the emojis along with their five skin tone variants all at once**
  * _person emojis will take longer to make, but they'll support skin tones right off the bat_
* **add just the basic yellow versions first, then do the skin tones later**
  * _the process for adding person emojis won't change, but it means a lot more work later_

This will probably need to be discussed before we can add support for skin tones.

**As for the GitHub emojis:** Forumoji is only adding emojis that exist in the Unicode standard. Any platform-specific or custom emojis are out of the question. GitHub-only emojis like octocat and trollface won't be added.
